compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -7270,4 +7270,8 @@ public boolean isContainSubtypeOfInt(BType type) {
                 return false;
         }
     }
+
+    public boolean isMappingConstructorCompatibleType(BType type) {
+        return Types.getReferredType(type).tag == TypeTags.RECORD || Types.getReferredType(type).tag == TypeTags.MAP;
```

Comment:
Since we have moved the method. Can refactor too. 
```suggestion
        int tag = getReferredType(type).tag;
        return tag == TypeTags.RECORD || tag == TypeTags.MAP;
```